addrindex panic

```
14:29:20 2016-09-16 [INF] INDX: Exists address index is enabled
14:29:20 2016-09-16 [INF] INDX: Catching up indexes from height 0 to 164286
14:29:33 2016-09-16 [INF] INDX: Indexed 179 blocks in the last 12.25s (279 transactions, height 179, 2016-01-27 12:08:19 -0500 EST)
14:29:46 2016-09-16 [INF] INDX: Indexed 1 block in the last 13.01s (21 transactions, height 180, 2016-01-27 12:08:22 -0500 EST)
14:29:58 2016-09-16 [INF] INDX: Indexed 39 blocks in the last 12.09s (823 transactions, height 219, 2016-01-27 12:14:40 -0500 EST)
14:30:12 2016-09-16 [INF] INDX: Indexed 8 blocks in the last 13.9s (168 transactions, height 227, 2016-01-27 12:15:22 -0500 EST)
14:30:23 2016-09-16 [INF] INDX: Indexed 23 blocks in the last 11.25s (485 transactions, height 250, 2016-01-27 12:18:31 -0500 EST)
14:30:30 2016-09-16 [INF] STKE: Storing the ticket database to disk
14:30:48 2016-09-16 [INF] DCRD: Gracefully shutting down the database...
14:30:49 2016-09-16 [INF] DCRD: Shutdown complete
panic: runtime error: index out of range [recovered]
        panic: runtime error: index out of range

goroutine 1 [running]:
panic(0xa7afa0, 0xc420010110)
        /home/dhill/git/go/src/runtime/panic.go:500 +0x1a1
github.com/decred/dcrd/database/ffldb.rollbackOnPanic(0xc43a368bd0)
        /home/dhill/.go/src/github.com/decred/dcrd/database/ffldb/db.go:1890 +0x77
panic(0xa7afa0, 0xc420010110)
        /home/dhill/git/go/src/runtime/panic.go:458 +0x243
github.com/decred/dcrd/blockchain/indexers.(*AddrIndex).ConnectBlock(0xc43dae1380, 0xf8e900, 0xc43a368bd0, 0xc43dd48200, 0xc43a911c80, 0xc43f513320, 0xc400000302, 0x0)
        /home/dhill/.go/src/github.com/decred/dcrd/blockchain/indexers/addrindex.go:861 +0x4cf
github.com/decred/dcrd/blockchain/indexers.dbIndexConnectBlock(0xf8e900, 0xc43a368bd0, 0xf8cba0, 0xc43dae1380, 0xc43dd48200, 0xc43a911c80, 0xc43f513320, 0xc43f249230, 0xc43f249220)
        /home/dhill/.go/src/github.com/decred/dcrd/blockchain/indexers/manager.go:90 +0x133
github.com/decred/dcrd/blockchain/indexers.(*Manager).Init.func5(0xf8e900, 0xc43a368bd0, 0xc43a368bd0, 0x0)
        /home/dhill/.go/src/github.com/decred/dcrd/blockchain/indexers/manager.go:446 +0x227
github.com/decred/dcrd/database/ffldb.(*db).Update(0xc42000ca00, 0xc43a671a90, 0x0, 0x0)
        /home/dhill/.go/src/github.com/decred/dcrd/database/ffldb/db.go:1947 +0xc0
github.com/decred/dcrd/blockchain/indexers.(*Manager).Init(0xc4356726f0, 0xc4200be3c0, 0x0, 0x0)
        /home/dhill/.go/src/github.com/decred/dcrd/blockchain/indexers/manager.go:455 +0x84f
github.com/decred/dcrd/blockchain.New(0xc42b449448, 0xc43defabd0, 0xc43def8f60, 0xc4203eddd0)
        /home/dhill/.go/src/github.com/decred/dcrd/blockchain/chain.go:2144 +0x773
main.newBlockManager(0xc4200be140, 0xf892c0, 0xc4356726f0, 0x2, 0x3, 0xc43dae1400)
        /home/dhill/.go/src/github.com/decred/dcrd/blockmanager.go:2919 +0x4a2
main.newServer(0xc420012fc0, 0x3, 0x3, 0xf8b8e0, 0xc42000ca00, 0xc4203d6300, 0xfcacc0, 0x20, 0xc4202e42d0, 0xc420014740)
        /home/dhill/.go/src/github.com/decred/dcrd/server.go:2609 +0x1644
main.dcrdMain(0x0, 0x0, 0x0)
        /home/dhill/.go/src/github.com/decred/dcrd/dcrd.go:205 +0xa29
main.main()
        /home/dhill/.go/src/github.com/decred/dcrd/dcrd.go:268 +0x66
```

--- a/blockchain/indexers/addrindex.go
+++ b/blockchain/indexers/addrindex.go
@@ -720,6 +720,7 @@ func (idx *AddrIndex) indexPkScript(data writeIndexData, scriptVersion uint16, p
 func (idx *AddrIndex) indexBlock(data writeIndexData, block, parent *dcrutil.Block, view *blockchain.UtxoViewpoint) {
 	regularTxTreeValid := dcrutil.IsFlagSet16(block.MsgBlock().Header.VoteBits,
 		dcrutil.BlockValid)
+	var stakeStartIdx int
 	if regularTxTreeValid {
 		for txIdx, tx := range parent.Transactions() {
 			// Coinbases do not reference any inputs.  Since the block is
@@ -753,9 +754,10 @@ func (idx *AddrIndex) indexBlock(data writeIndexData, block, parent *dcrutil.Blo
 					false)
 			}
 		}
+
+		stakeStartIdx = len(parent.Transactions())
 	}
 
-	stakeStartIdx := len(parent.Transactions())
 	for txIdx, tx := range block.STransactions() {
 		thisTxOffset := txIdx + stakeStartIdx
 

